### PR TITLE
Unlock PHP 8 + Contao 4.1x

### PR DESCRIPTION
**Pull request by rabauss**
Support for php 8.
Fix compatiblitiy with contao > 4.10.

### CODE REVIEW

**Comment by rabauss** [comment]
@koertho any suggestions to merge this PR?

**Comment by koertho** [comment]
@rabauss Sorry, had busy days. I'll try to have a look into it today!

File: src/Resources/contao/dca/tl_calendar_events.php [hunk 1]
```diff
@@ -6,21 +6,25 @@
...
-    'show', [
-    'feature' => [
-        'label'           => &$GLOBALS['TL_LANG']['tl_calendar_events']['feature'],
-        'icon'            => 'featured.svg',
-        'attributes'      => 'onclick="Backend.getScrollOffset();return AjaxRequest.toggleFeatured(this,%s)"',
-        'button_callback' => [\HeimrichHannot\EventsBundle\EventListener\DataContainer\CalendarEventsListener::class, 'iconFeatured'],
-    ],
-], 0
-);
+if (version_compare(PackageUtil::getContaoVersion(), '4.10', '<')) {
```
Comment (koertho) [inline src/Resources/contao/dca/tl_calendar_events.php:15]: Could you put this code into a load callback or loadDataContainer hook? Services should not be loaded in a dca file.

**Review by koertho — CHANGES_REQUESTED** [review]
Thank you very much @rabauss ! I only have one point where we can enhance the code here, would be great if you could realize it.

File: src/Resources/contao/dca/tl_calendar_events.php (hunk 1 above)
Comment (rabauss) [reply src/Resources/contao/dca/tl_calendar_events.php:15]: Sorry, but I also don't have the time at the moment. But there are a lot more service usages in dca files in this extension. I only added the version check for compatibility with 4.x in this PR!

Comment (koertho) [reply src/Resources/contao/dca/tl_calendar_events.php:15]: I completely unterstand this. Than I will merge it and maybe find some time to refactor later.

**Comment by koertho** [comment]
It's available in version 1.8.0. Thanks @rabauss !

**Pull request merged**